### PR TITLE
Added unittest for date and datetime

### PR DESCRIPTION
**Pull request by Flyingmana**

### CODE REVIEW

File: dev/tests/unit/Varien/Data/Form/FilterTest.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        );
+        $this->assertEquals(
+            '0090-05-18',
+            $subject->inputFilter('90-05-18')
+        );
+        $this->assertEquals(
+            '1990-05-08',
+            $subject->inputFilter('1990-5-8')
+        );
+
```
Comment (kiatng) [inline dev/tests/unit/Varien/Data/Form/FilterTest.php:36]:
```suggestion
        $this->assertEquals(
            '1970-01-01',
            $subject->inputFilter('1970-01-01')
        );
   
```

**Comment by fballiano** [comment]
I've no idea how to test this PR, is it running in some workflow?

**Comment by Flyingmana** [comment]
it runs in the sonar workflow https://github.com/OpenMage/magento-lts/blob/e08d88941cff9fd51b0526d39f57ca595ed544f9/.github/workflows/sonar.yml#L48

**Review by fballiano — APPROVED** [review]
tested, runs alright

**Review by ADDISON74 — APPROVED** [review]

**Pull request merged**